src/NServiceBus.Transport.SQS/InputQueuePump.cs

```diff
@@ -288,6 +293,11 @@ internal async Task ProcessMessage(Message receivedMessage, CancellationToken me
                 isPoisonMessage = true;
             }
 
+            if (messagesToBeDeleted.TryGet(messageId, out _))
```

Comment:
We should move this check to directly after the `messageId` value is assigned. We don't care about the `s3BodyKey` value at this point because if the message is in `messagesToBeDeleted` we've already called `DeleteMessage` once and logged the message if the body is in S3.

Reply:
Also, we should `return` if we find this message because we want to skip all additional processing as this point.